api/src/main/java/com/valtech/aem/saas/api/bestbets/BestBetsService.java

```diff
@@ -0,0 +1,17 @@
+package com.valtech.aem.saas.api.bestbets;
+
+/**
+ * Represents a service that manages best bets.
+ */
+public interface BestBetsService {
+
+  /**
+   * Returns a best bets consumer object that provides actions for consuming/managing best bets, for a specified client
+   * identifier.
+   *
+   * @param client identifying string.
+   * @return best bets consumer object
+   * @throws IllegalArgumentException exception thrown when blank client argument is passed
+   */
+  BestBetsConsumerService getBestBetsConsumerService(String client);
```

Comment:
- [ ] please add exception throing to the method signature

Reply:
> * [ ]  please add exception throing to the method signature
at the time, i googled for what would the best practice be, for this situation, and found this thread:
https://stackoverflow.com/questions/824217/should-methods-that-throw-runtimeexception-indicate-it-in-method-signature

Reply:
> I see, this should be handled as the index and be a context aware configuration than

Field for client is already added in SearchConfiguration. If you refer to the simpified interface, that i'd do in another PR